packaging/hudi-aws-bundle/pom.xml

```diff
@@ -0,0 +1,264 @@
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>hudi</artifactId>
+        <groupId>org.apache.hudi</groupId>
+        <version>0.12.0-SNAPSHOT</version>
+        <relativePath>../../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>hudi-aws-bundle</artifactId>
+    <packaging>jar</packaging>
+
+    <properties>
+        <checkstyle.skip>true</checkstyle.skip>
+        <main.basedir>${project.parent.basedir}</main.basedir>
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.rat</groupId>
+                <artifactId>apache-rat-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <version>${maven-shade-plugin.version}</version>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <createSourcesJar>${shadeSources}</createSourcesJar>
+                            <dependencyReducedPomLocation>${project.build.directory}/dependency-reduced-pom.xml
+                            </dependencyReducedPomLocation>
+                            <transformers>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheLicenseResourceTransformer">
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ApacheNoticeResourceTransformer">
+                                    <addHeader>true</addHeader>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.IncludeResourceTransformer">
+                                    <resource>META-INF/LICENSE</resource>
+                                    <file>target/classes/META-INF/LICENSE</file>
+                                </transformer>
+                                <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+                            </transformers>
+                            <artifactSet>
+                                <includes>
```

Comment:
Not sure I understand the need to bundle all these dependencies in the `aws` bundle. In my understanding, this bundle should contain only AWS specific dependencies required for AWS specific features like cloudwatch metrics, dynamodb locking etc to work, so I would assume the following should suffice:
- hudi-aws
- sdk's
- dyamodb-lock-client

I would assume that this would need to be used in conjunction with `hudi-spark-bundle` which already should have other dependencies bundled in it. By bundling them here as well, we are duplicating these dependencies which does not seem necessary.

Reply:
@umehrot2 we're mostly aligned. but for aws-bundle, we should at least be able to run `AwsGlueCatalogSyncTool` with the bundle jar alone. Similarly, hudi-gcp-bundle can also run `BigQuerySyncTool`. In short, if a bundle jar contains runnable main class, users should be able to run it out-of-box without additional dependencies.

Reply:
Currently `AwsGlueCatalogSyncTool` piggybacks on `HiveSyncTool` hence we need to include hive sync dependencies. In future improvements, we should consider make aws sync independent of hive sync libs.

Reply:
Agree. HUDI-4169 to track it.
Meanwhile, I verified the glue sync with the new bundle and it works as shown in the screenshot below.

Reply:
I see. In that case, yeah we need to include those dependencies needed to make the `AwsGlueCatalogSyncTool` run independently. But we should try to make it work without needed hive sync libs preferably. Thanks for filing the jira.